.cirrus.yml

```diff
@@ -0,0 +1,227 @@
+# Reference:
+#   - https://cirrus-ci.org/guide/writing-tasks/
+#   - https://cirrus-ci.org/guide/tips-and-tricks/#sharing-configuration-between-tasks
+#   - https://cirrus-ci.org/guide/linux/
+#   - https://cirrus-ci.org/guide/macOS/
+#   - https://cirrus-ci.org/guide/windows/
+#   - https://hub.docker.com/_/gcc/
+#   - https://hub.docker.com/_/python/
+
+#
+# Global defaults.
+#
+container:
+  image: python:3.8
+  cpu: 2
+  memory: 4G
+
+
+env:
+  # Maximum cache period (in weeks) before forcing a new cache upload.
+  CACHE_PERIOD: "2"
+  # Increment the build number to force new cartopy cache upload.
+  CARTOPY_CACHE_BUILD: "0"
+  # Increment the build number to force new conda cache upload.
+  CONDA_CACHE_BUILD: "0"
+  # Increment the build number to force new nox cache upload.
+  NOX_CACHE_BUILD: "0"
+  # Increment the build number to force new pip cache upload.
+  PIP_CACHE_BUILD: "0"
+  # Pip package to be upgraded/installed.
+  PIP_CACHE_PACKAGES: "pip setuptools wheel nox"
+  # Git commit hash for iris test data.
+  IRIS_TEST_DATA_REF: "fffb9b14b9cb472c5eb2ebb7fd19acb7f6414a30"
+  # Base directory for the iris-test-data.
+  IRIS_TEST_DATA_DIR: ${HOME}/iris-test-data
+
+
+#
+# Linting
+#
+lint_task:
+  auto_cancellation: true
+  name: "Lint: flake8 and black"
+  pip_cache:
+    folder: ~/.cache/pip
+    fingerprint_script:
+      - echo "${CIRRUS_OS} py${PYTHON_VERSION}"
+      - echo "$(date +%Y).$(($(date +%U) / ${CACHE_PERIOD})):${PIP_CACHE_BUILD} ${PIP_CACHE_PACKAGES}"
+  lint_script:
+    - pip list
+    - python -m pip install --retries 3 --upgrade ${PIP_CACHE_PACKAGES}
+    - pip list
+    - nox --session flake8
+    - nox --session black
+
+
+#
+# YAML alias for common linux test infra-structure.
+#
+linux_task_template: &LINUX_TASK_TEMPLATE
```

Comment:
Is this ampersand correct?

Reply:
Yup. See https://cirrus-ci.org/guide/tips-and-tricks/#sharing-configuration-between-tasks

Essentially, it's a mechanism to use YAML aliases in order to share common YAML in several places. It's pretty cool.

In summary, there are two parts to this. The `&` part and the `*` part below... it's kinda like jinja templating, for template definition (`&`) and then template use (`*`).

Does that help?